### ISSUE TITLE
Add sort parameter to list_users

### ISSUE THREAD

**Issue by perenstrom** [emreakay/CodeIgniter-Aauth#176]
It would be nice with a parameter to the `list_users` function to be able to specify which field to sort by. Useful if you want to return users according to username.

**Comment by emreakay** [comment]
thank you for your advice @perenstrom. It is nice idea but you can also do this by using array sort functions in php.

**Comment by REJack** [comment]
@perenstrom nice idea.
@emreakay thats right but it costs performance :smile:

**Comment by perenstrom** [comment]
@emreakay, I get back an array of objects right? How can I sort that with array sort functions? Say I would want to sort by users->username?

**Comment by perenstrom** [comment]
And precisely as @REJack says, sorting directly in database call would be very much faster.

**Comment by REJack** [comment]
FYI:

**Comment by REJack** [comment]
Hi guys i tested it :smile:

Here my results:

default list_users()
- User count: 768. Page rendered in 1.0730 seconds. CodeIgniter Version 3.1.0

array_multisort()
- User count: 768. Page rendered in 1.1956 seconds. CodeIgniter Version 3.1.0

tweaked list_users()
- User count: 768. Page rendered in 1.0980 seconds. CodeIgniter Version 3.1.0

@perenstrom for a quick solution:
```php
        $users = $this->aauth->list_users();
        foreach ($users as $key => $user) {
            $_sortfield[$key] = strtolower($user->name);
        }
        array_multisort($_sortfield, SORT_ASC, $users);
        print_r($users);
```

**Comment by perenstrom** [comment]
Okay, so not much performance problems. However:
        $users = $this->aauth->list_users();
        foreach ($users as $key => $user) {
            $_sortfield[$key] = strtolower($user->name);
        }
        array_multisort($_sortfield, SORT_ASC, $users);
        print_r($users);
is still uglier than 
    $users = $this->aauth->list_users('name');

**Comment by REJack** [comment]
yeah i know, but i think on small hosting subscriptions it could make more difference.

I'll add this little feature in the next days.

my tweaked version looks a bit uglier than yours but its updatable for other users :smile:  
`$users = $this->aauth->list_users(FALSE, FALSE, FALSE, FALSE, 'name');`

**Comment by REJack** [comment]
I close this issue 😄 
Enhancement is released with [v2.5.12](https://github.com/emreakay/CodeIgniter-Aauth/releases/tag/v2.5.12)
